libs/utils/include/celix_errno.h

```diff
@@ -61,6 +73,44 @@ typedef int celix_status_t;
  */
 const char* celix_strerror(celix_status_t status);
 
+/*!
+ * Customer error code mask
+ *
+ */
+#define CELIX_CUSTOMER_ERR_MASK 0x20000000
+
+/*!
+ * The facility of system error code,
+ * \note Error code 0 indicates success,it is not system error code.
+ */
+#define CELIX_FACILITY_SYSTEM 0
+
+/*!
+ * The facility of celix default error code
+ *
+ */
+#define CELIX_FACILITY_NULL 1
```

Comment:
I am not sure why a `CELIX_FACILITY_NULL` is needed, is the `CELIX_FACILITY_SYSTEM` not a more  logical default facility error value?

Reply:
In the case of the existing error code value unchanged, I would like to make a distinction between celix default error code and C errno by using `CELIX_FACILITY_NULL` and `CELIX_FACILITY_SYSTEM`(To avoid ambiguity,I will rename `CELIX_FACILITY_SYSTEM` to `CELIX_FACILITY_CERRNO`). And we can replace existing error codes like following:
~~~C
#define CELIX_FACILITY_CERRNO 0
#define CELIX_FACILITY_NULL 1
const celix_status_t  CELIX_FRAMEWORK_EXCEPTION = CELIX_ERROR_MAKE(CELIX_FACILITY_NULL,   4471);//=70007
const celix_status_t  CELIX_FILE_IO_EXCEPTION= CELIX_ERROR_MAKE(CELIX_FACILITY_NULL,   4472);//=70008

const celix_status_t  CELIX_ENOMEM = CELIX_ERROR_MAKE(CELIX_FACILITY_CERRNO,   ENOMEM);//=ENOMEM
const celix_status_t  CELIX_EAGAIN = CELIX_ERROR_MAKE(CELIX_FACILITY_CERRNO,   EAGAIN);//=EAGAIN
~~~

Reply:
I like the rename to `CELIX_FACILITY_CERRNO`, that makes it more clear for me. 

I am still not sure about the `CELIX_FACILITY_NULL` name; Could this be renamed to `CELIX_FACILITY_FRAMEWORK`.

Than we have `CELIX_FACILITY_CERRNO` to communicate errno as-is and `CELIX_FACILITY_FRAMEWORK` for Celix framework errors.

Reply:
Thank you very much for your advice,I like  to  rename `CELIX_FACILITY_NULL` to `CELIX_FACILITY_FRAMEWORK`